src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -159,18 +173,23 @@ internal override GeneratedLabelSymbol BreakLabel
 
         protected override ImmutableArray<LabelSymbol> BuildLabels()
         {
-            ArrayBuilder<LabelSymbol> labels = null;
+            // We bind the switch expression and the switch case label expressions so that the constant values can be
+            // part of the label, but we do not report any diagnostics here. Diagnostics will be reported during binding.
 
+            ArrayBuilder<LabelSymbol> labels = ArrayBuilder<LabelSymbol>.GetInstance();
+            DiagnosticBag tempDiagnosticBag = DiagnosticBag.GetInstance();
```

Comment:
It might. I don't have any reason to believe that adding that code would have a measurable impact in the running time of the compiler.